controllers/ovndbcluster_controller.go

```diff
@@ -576,11 +606,78 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 				instance,
 				svcLabels,
 			)
+			if err != nil {
+				err = fmt.Errorf("Error while deleting service with name %s: %w", serviceName, err)
+				return ctrl.Result{}, err
+			}
+			// Delete DNS records for deleted services/pods
+			namespace := helper.GetBeforeObject().GetNamespace()
+			dnsName := fmt.Sprintf("%s-%s-%d", v1beta1.DNSPrefix, serviceName, i)
+			err = deleteDNSData(ctx, helper, dnsName, namespace)
 			if err != nil {
 				return ctrl.Result{}, err
 			}
 		}
 	}
+
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		err = fmt.Errorf("Error while retrieving OVNDBPods on %s cluster: %w", instance.Spec.DBType, err)
+		return ctrl.Result{}, err
+	}
+	// Assuming that the replicas is greater than 0 AND networkAttachment != ""
+	// dbAddress will contain ovsdbserver-(nb|sb).openstack.svc
+	// if replicas == 0 or networkAttachment == "" dbAddress will be empty.
+	// Meaning that Status.DBAddress is not ready.
+	dbAddress := ""
+
+	// DNSData is needed to provide DNS outside the cluster (edpm nodes) which needs to have
+	// spec.NetworkAttachment != ""
+	if instance.Spec.NetworkAttachment != "" {
+		for _, ovnPod := range podList.Items[:*(instance.Spec.Replicas)] {
+			// Get Hostname
+			svc, err := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			if err != nil {
+				err = fmt.Errorf("Error retrieving service %s: %w", ovnPod.Name, err)
+				return ctrl.Result{}, err
+			}
+
+			// Get IP
+			dnsIP, err := getPodIPv4InNetwork(ovnPod, instance)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			if len(dnsIP) == 0 {
+				Log.Info(fmt.Sprintf("Error while getting pod %s %s IPv4, IP is empty", ovnPod.Name, instance.Spec.NetworkAttachment))
+				continue
+			}
+
+			// Create DNSData CR
+			err = ovndbcluster.DnsData(
+				ctx,
+				helper,
+				svc,
+				dnsIP,
+				instance,
+				ovnPod,
+				serviceLabels,
+			)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			if dbAddress == "" {
+				dbAddress = ovndbcluster.GetDBAddress(instance, svc)
```

Comment:
(nit) it would be slightly more clean if you would consolidate the code that sets DBAddress in a single code block, as in https://github.com/booxter/ovn-operator/commit/deacd61b39ec906617ba606cebedf98d89cdb66e

Reply:
I like it, done

Reply:
Need to add a check in GetDBAddress bc if NAD == "" svc will be null